pkg/auth/authproviders/oidc/backend_impl.go

```diff
@@ -204,9 +206,15 @@ func (p *backendImpl) fetchUserInfo(ctx context.Context, rawAccessToken string)
 		return nil, errors.Wrap(err, "extracting external claims from userinfo endpoint response")
 	}
 
+	rawUserInfo, err := claimsAsString(userInfoFromEndpoint)
+	if err != nil {
+		return nil, errors.Wrap(err, "extracting raw user info from userinfo endpoint response")
+	}
+
 	return &authproviders.AuthResponse{
-		Claims:     externalClaims,
-		Expiration: time.Now().Add(userInfoExpiration),
+		Claims:          externalClaims,
+		Expiration:      time.Now().Add(userInfoExpiration),
+		UnderlyingToken: rawUserInfo,
```

Comment:
I think the best in two cases would be to get the claims as a JSON string from either the raw ID token or the user info response.

This way we don't have this indirection of potentially having a JWT token string _or_ a JSON string of the claims in test mode depending on whether the refresh or access token is used; it also wouldn't be understandable why the same call either responds with a JWT token or with a JSON string of claims.

Since the claims are really what we are interested in w.r.t. the underlying ID token, this should be sufficient enough.